src/context/cart-context.tsx

```diff
@@ -16,13 +16,12 @@ export interface CartContextInterface {
   addProductToCart: (
     product: ProductObjectInCartModel
   ) => Promise<CartModel | undefined>;
-  cart: CartModel | undefined;
-  updateProductToCart: (
-    product: ProductObjectInCartModel
-  ) => Promise<CartModel | undefined>;
-  clearProductsInCart: () => Promise<CartModel | undefined>;
+  cart?: CartModel;
+  updateProductToCart: (product: ProductObjectInCartModel) => Promise<void>;
+  clearProductsInCart?: () => Promise<CartModel | undefined>;
   deleteProductInCart: (productUuid: string) => Promise<CartModel | undefined>;
-  contextValue: ContextValueInterface;
+  isActive: boolean;
+  setIsActive: Dispatch<React.SetStateAction<boolean>>;
 }
 export interface PropsContextModel {
```

Comment:
React has already defined Interface for children `ChildrenWithProps` google that and apply :)